src/main/java/chess/domain/attribute/Square.java

```diff
@@ -0,0 +1,232 @@
+import java.util.Objects;
+
+public class Square {
+
+    private static final Map<String, Square> SQUARES = new HashMap<>();
+
+    static {
+        initializeSquares();
+    }
+
+    private final File file;
+    private final Rank rank;
+
+    private Square(final File file, final Rank rank) {
+        this.file = file;
+        this.rank = rank;
+    }
+
+    public static Square of(final File file, final Rank rank) {
+        return SQUARES.get(keyOf(file, rank));
+    }
+
+    private static void initializeSquares() {
+        for (final Rank rank : Rank.values()) {
+            putSquares(rank);
+        }
+    }
+
+    private static void putSquares(final Rank rank) {
+        for (final File file : File.values()) {
+            SQUARES.put(keyOf(file, rank), new Square(file, rank));
+        }
+    }
+
+    private static String keyOf(final File file, final Rank rank) {
+        return file.name() + rank.name();
+    }
+
+    public boolean canMove(Movement movement) {
+        return canMoveHorizontal(movement.getX()) && canMoveVertical(movement.getY());
+    }
+
+    public boolean canMoveVertical(int step) {
+        if (step > 0) {
+            return canMoveUp(Math.abs(step));
+        }
+        if (step < 0) {
+            return canMoveDown(Math.abs(step));
+        }
+        return true;
+    }
+
+    public boolean canMoveHorizontal(int step) {
+        if (step > 0) {
+            return canMoveRight(Math.abs(step));
+        }
+        if (step < 0) {
+            return canMoveLeft(Math.abs(step));
+        }
+        return true;
+    }
+
+    public Square move(Movement movement) {
+        return moveHorizontal(movement.getX()).moveVertical(movement.getY());
+    }
+
+    public Square moveHorizontal(int step) {
+        if (step < 0) {
+            return moveLeft(Math.abs(step));
+        }
+        if (step > 0) {
+            return moveRight(Math.abs(step));
+        }
+        return this;
+    }
+
+    public Square moveVertical(int step) {
+        if (step < 0) {
+            return moveDown(Math.abs(step));
+        }
+        if (step > 0) {
+            return moveUp(Math.abs(step));
+        }
+        return this;
+    }
+
+    public Square moveUp(int step) {
+        return Square.of(file, rank.up(Math.abs(step)));
+    }
+
+    public Square moveUp() {
+        return Square.of(file, rank.up());
+    }
+
+    public Square moveDown(int step) {
+        return Square.of(file, rank.down(Math.abs(step)));
+    }
+
+    public Square moveDown() {
+        return Square.of(file, rank.down());
+    }
+
+    public Square moveLeft(int step) {
+        return Square.of(file.left(Math.abs(step)), rank);
+    }
+
+    public Square moveLeft() {
+        return Square.of(file.left(), rank);
+    }
+
+    public Square moveRight(int step) {
+        return Square.of(file.right(Math.abs(step)), rank);
+    }
+
+    public Square moveRight() {
+        return Square.of(file.right(), rank);
+    }
```

Comment:
오버라이드 코드끼리는 가장 메인이 되는 메서드를 정해보셔요. 이러면 가장 메인인 메서드만 테스트되도 좀 안심이겠죠.
```suggestion
    public Square moveRight(int step) {
        return Square.of(file.right(Math.abs(step)), rank);
    }

    public Square moveRight() {
        return moveRight(1);
    }
```